### PR TITLE
Implement periodic gathering as a job in tech preview

### PR DESCRIPTION
**Pull request by tremes**
This changes the periodic data gathering to run as a Kubernetes jobs. This includes using and implementing a new `datagather.insights.openshift.io` custom resource. A new `NewGatherAndUpload` command was added to `pkg/cmd/start/start.go`, but it's really implemented in `pkg/controller/gather_commands.go` (which is not really new, but renamed from `gather_job.go`. 

The Kubernetes job is created in the `pkg/controller/periodic/job.go` and is responsible for the following:

- run the data gathering
- recording the data as an archive  - this is the change in `pkg/recorder/diskrecorder/diskrecorder.go` - we are only interested in the last archive (simply because there won't be more)
- uploading the data - these are the changes in `pkg/insights/insightsuploader/insightsuploader.go` and `pkg/insights/insightsclient/requests.go`, because we need to know and use the `insightsRequestID`

Job is executed periodically by the new method ` GatherJob()` in `pkg/controller/periodic/periodic.go`. Each job also requires a new `DataGather` CR (created in  `pkg/controller/periodic/periodic.go` by method `createNewDataGatherCR(ctx context.Context, disabledGatherers []string,	dataPolicy insightsv1alpha1.DataPolicy)`). This custom resource reflects the same values, which user can configure in `insightsdatagather.config.openshift.io` resource) and this is the responsibility of the `createDataGatherAttributeValues() ([]string, insightsv1alpha1.DataPolicy)` method in  `pkg/controller/periodic/periodic.go`. 

Finally the gatherers status needs to "copied" from the corresponding `DataGather` resource to the `insightsoperator.operator.openshift.io` resource - this is done in `copyDataGatherStatusToOperatorStatus(ctx context.Context, dgName string) (*v1.InsightsOperator, error) ` in `pkg/controller/periodic/periodic.go`. 

## Categories

- [ ] Bugfix
- [ ] Data Enhancement
- [X] Feature
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive

No new data

## Documentation

No docs update yet

## Unit Tests

- `pkg/controller/periodic/job_test.go` - a new test
- `pkg/controller/status/gatherer_status_test.go` - a new test
- `pkg/controller/periodic/periodic_test.go` - updated and extended
- `pkg/gather/gather_test.go` - updated
- 
## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes/No

## References

https://issues.redhat.com/browse/???
https://access.redhat.com/solutions/???

### CODE REVIEW

File: pkg/controller/gather_commands.go [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+	dataGatherCR.Status.State = insightsv1alpha1.Completed
+	dataGatherCR.Status.InsightsRequestID = insightsRequestID
+	dataGatherCR.Status.Conditions = []metav1.Condition{
+		{
+			Type:               "DataUploaded",
+			Status:             metav1.ConditionTrue,
+			Reason:             "AsExpected",
+			LastTransitionTime: metav1.Now(),
+		},
+	}
```
Comment (tremes) [inline pkg/controller/gather_commands.go:217]: This status updating is temporary and will be addressed in https://issues.redhat.com/browse/CCXDEV-10590

**Comment by rluders** [comment]
I want to clarify one thing about the jobs first.
/remove-lgtm

File: pkg/controller/periodic/job.go [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+// derived from the provided data gather name
+func (j *JobController) CreateGathererJob(ctx context.Context, dataGatherName, image, archiveVolumeMountPath string) (*batchv1.Job, error) {
+	gj := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      dataGatherName,
+			Namespace: insightsNamespace,
+		},
+		Spec: batchv1.JobSpec{
+			// backoff limit is 0 - we dont' want to restart the gathering immediately in case of failure
+			BackoffLimit: new(int32),
```
[the remark this answers is not in the document]
Comment (tremes) [reply pkg/controller/periodic/job.go:38]: Do you mean this https://github.com/kubernetes/api/blob/v0.27.1/batch/v1/types.go#L302 ? Yes we keep the finished jobs on purpose and that's why there's this https://github.com/openshift/insights-operator/pull/764/files#diff-dec12f0cf28a91a837b46e755b46c081c9014121cdb770b2cd8ac916af29c8deR393 pruning method. Jobs older than 24h are removed....so I think there's currently no reason to set the TTL attribute.

[the remark this answers is not in the document]
Comment (tremes) [reply pkg/controller/periodic/job.go:38]: User is not supposed to create or run the job manually. Creating the `periodic-gather` jobs is the operator responsibility.

[the remark this answers is not in the document]
Comment (rluders) [reply pkg/controller/periodic/job.go:38]: OK, I tested it again, and it is working as expected. I think that the cluster that I was playing with was broken.

**Comment by DennisPeriquet** [comment]
manually tagging this PR for "Feature Merge Queue"

**Comment by tremes** [comment]
yeah I just found out that. Sorry. Let me revert it

**Pull request merged**